.github/workflows/codeql.yml

```diff
@@ -0,0 +1,46 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [dev]
+  pull_request:
+    branches: [dev]
```

Comment:
do we do like all other workflows and keep it for all PRs regardless of the target branch?

Reply:
I'll keep this for now and I think we need to discuss it.
Right now I don't think we need to run codeql check for all the branches. It will be enough to check open PRs and dev.

Reply:
sure thing